src/sentry/rules/history/preview_strategy.py

```diff
@@ -0,0 +1,83 @@
+from typing import Any, Dict, Sequence
+
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.events import Column
+from sentry.types.issues import GroupCategory
+
+"""
+Issue category specific components to computing a preview for a set of rules.
+
+To add support for a new issue category/dataset:
+    1. Add mapping from GroupCategory to Dataset
+    2. Add mapping from Dataset to snuba column name
+        a. The column name should be a field in sentry.snuba.events.Column
+    3. Add category-specific query params for GROUPS_STRATEGIES and GROUP_STRATEGIES
+"""
+
+# Maps group category to dataset
+GROUP_CATEGORY_TO_DATASET: Dict[GroupCategory, Dataset] = {
+    GroupCategory.ERROR: Dataset.Events,
+    GroupCategory.PERFORMANCE: Dataset.Transactions,
+}
+
+# Maps datasets to snuba column name
+DATASET_TO_COLUMN_NAME: Dict[Dataset, str] = {
+    Dataset.Events: "event_name",
+    Dataset.Transactions: "transaction_name",
+}
+
+
+# Given a list of Column Enum objects, return their actual column name in each dataset that is supported
+def get_dataset_columns(columns: Sequence[Column]) -> Dict[Dataset, Sequence[str]]:
+    dataset_columns: Dict[Dataset, Sequence[str]] = {}
+    for dataset, column_name in DATASET_TO_COLUMN_NAME.items():
+        dataset_columns[dataset] = [
+            getattr(column.value, column_name)
+            for column in columns
+            if getattr(column.value, column_name) is not None
+        ]
+
+    return dataset_columns
+
+
+def _events_kwargs(group_ids: Sequence[int], kwargs: Dict[str, Any]) -> Dict[str, Any]:
+    kwargs["conditions"] = [("group_id", "IN", group_ids)]
+    return kwargs
+
+
+def _transactions_kwargs(group_ids: Sequence[int], kwargs: Dict[str, Any]) -> Dict[str, Any]:
+    kwargs["having"] = [("group_id", "IN", group_ids)]
```

Comment:
Nice use of having clause. I ended up using `arrayIntersect` to do something similar: https://github.com/getsentry/sentry/pull/41685/files#diff-7cb5118757db3a39c7883599b74b268712e8493868db9ddde398aea2b3136f5cR163-R167

but `HAVING` is way more elegant.